main/src/main/scala/sbt/Project.scala

```diff
@@ -119,7 +119,13 @@ sealed trait ProjectDefinition[PR <: ProjectReference] {
     if (ts.isEmpty) Nil else s"$label: $ts" :: Nil
 }
 
-sealed trait Project extends ProjectDefinition[ProjectReference] {
+trait CompositeProject {
```

Comment:
could you seal this please?

Reply:
In the implementation of, say, `CrossProject`, we would want to extend this. In the sbt-test, I just create an instance, but in reality, I _think_ we would want to extend. 

Of course, I may have got the use wrong. But my impression was that by doing this, its then possible to have a `CrossProject` as a „real“ `Project`

Reply:
Hm... maybe I‘m wrong, sigh. OK 👍

Reply:
Er ... @BennyHill is right, here. If it's sealed, this thing is completely useless ;) The whole point of this thing is that we can indeed have `CrossProject` implement that trait (which is a pure Java interface, btw). Otherwise, how do you propose we leverage this in sbt-crossproject?

Reply:
OK.... How about that when I do the other changes, I update the tests to mimick invisaged usage. That way, it^s clear-cut

Reply:
> what if we provide a ProjectsList subclass?

What exactly does that buy you, if that one is open, compared to having the trait open? The only thing I see is that you would be able to add a concrete field in a binary compatible way, but ... that would be weird anyway.

For things that build on top of it, it's a limitation, as they cannot extend `ProjectsList` *and* another class, but they could extend `CompositeProject` and a class, because `CompositeProject` is an interface.

Reply:
> What exactly does that buy you, if that one is open

it wouldn't be open `final class ProjectsList` that extends `CompositeProject` I was thinking.

Reply:
Then I don't see how I can use it in `CrossProject` either, since I need `crossProject` to return something that is both a `CompositeProject` (for sbt to pick it up as such) *and* have methods that are specific to `CrossProject`s (so that I can do `.dependsOn(foo.jvm)` in some other JVM project, at least. If my only options to have a `CompositeProject` are `Project` which is sealed and `ProjectsList` which is final, it's no good.

Basically, think about how I can declare `CrossProject` so that the following typechecks and is identified as a `CompositeProject`:
```scala
lazy val foo = crossProject
  .settings(...)
  .jvmSettings(...)
  .jsSettings(...)

lazy val bar = project
  .settings(...)
  .dependsOn(foo.jvm)
```

Reply:
With an open `CompositeProject`, it's super easy. I just alter the [declaration of `CrossProject`](https://github.com/portable-scala/sbt-crossproject/blob/f1a6e1c7c18f109599174a4f8cfad60e1afb9a21/sbt-crossproject/src/main/scala/sbtcrossproject/CrossProject.scala#L10) with `extends CompositeProject` and
```scala
def componentProjects: Seq[Project] = projects.values.toSeq
```

Reply:
got it. thanks for detailing the use case.